snapcraft/extractors/_metadata.py

```diff
@@ -63,7 +73,23 @@ def get_description(self) -> str:
         """
         return self._data.get('description')
 
-    def to_dict(self) -> Dict[str, str]:
+    def get_icon(self) -> str:
+        """Return extracted icon.
+
+        :returns: Extracted icon
+        :rtype: str
+        """
+        return self._data.get('icon')
+
+    def get_desktop(self) -> List[str]:
```

Comment:
I'm finding the singular confusing here since there can be several files. Can we make that `get_desktops` or `get_desktop_files`?

Reply:
I much prefer the "desktop files" language.

Reply:
The thing is that these are not desktop_files. They are desktop_file_ids, as defined in https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
I kind of wanted not to specify too much in the name, because other sources of metadata might user file paths instead of ids, or file names, or something else. But well, this is not clear, I agree with you.

I will rename it get_destkop_file_ids. And later, if there's another type, we can add an attribute and a get_desktop_file_paths, or something like that. It won't be nice and consistent, but that seems unavoidable given this weird desktop_file_id thing.

Reply:
Aha! Thanks for elaborating on that. In that case `get_desktop_file_ids` sounds perfect.